### PR TITLE
#167532588 Fix the return error on create client company

### PR DESCRIPTION
**Pull request by shemaeric**
## Description ##
Return the meaningful error messages while creating a new client company without data provided by the user. 

## Type of change ##
Please select the relevant option
- [x] Bug fix(a non-breaking change which fixes an issue)
- [ ] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [ ] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#167532588](https://www.pivotaltracker.com/n/projects/2343772/stories/167532588)

### CODE REVIEW

File: transactions/transaction_services.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
 class TransactionServices:
     """A helper class for online payment services"""
 
-    rave_public_key = os.getenv('RAVE_PUBLIC_KEY')
-    rave_secret_key = os.getenv('RAVE_SECRET_KEY')
+    rave_public_key = os.getenv('RAVE_TEST_PUBLIC_KEY')
```
Comment (espoirMur) [inline transactions/transaction_services.py]: Bi bro, we have agreed to use RAVE_PUBLIC_KEY as variable names, 
Kindly change them in your .env file

File: tests/authentication/client/test_client_company.py [hunk 2]
```diff
@@ -178,7 +193,7 @@ def test_get_client_company(self, mock_email):
         """Get client a company account."""
         self.set_token()
         mock_email.return_value = True
-        self.client.post(
+        postRes = self.client.post(
```
Comment (espoirMur) [inline tests/authentication/client/test_client_company.py]: use snake case conventions for variable names, we are in python here! lool

**Review by espoirMur — CHANGES_REQUESTED** [review]

File: transactions/transaction_services.py (hunk 1 above)
Comment (shemaeric) [reply transactions/transaction_services.py]: Thank you @espoirMur

**Comment by MandelaK** [comment]
I have tried to look at why there is a coverage drop, but Coveralls restricts me ... Did you test all the functionality you introduced to this PR? 0.6% drops is quite large

File: transactions/transaction_services.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class TransactionServices:
     """A helper class for online payment services"""
 
     rave_public_key = os.getenv('RAVE_PUBLIC_KEY')
-    rave_secret_key = os.getenv('RAVE_SECRET_KEY')
+    rave_secret_key = os.getenv('RAVE_TEST_SECRET_KEY')
```
Comment (Victoradukwu) [inline transactions/transaction_services.py]: @shemaeric The right name of the environment variable is `RAVE_SECRET_KEY`. Do not override it.

File: authentication/views.py [hunk 4]
```diff
@@ -246,6 +246,9 @@ def get(self, request):
 
     def post(self, request):
         data = request.data
+        serializer = self.serializer_class(data = request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
```
Comment (Victoradukwu) [inline authentication/views.py]: @shemaeric Why are we calling `serializer.save()` here, since we have it on line **271**?

**Review by Victoradukwu — CHANGES_REQUESTED** [review]
@shemaeric Thanks for this awesome job. Kindly attend to the requested changes and the decrease in test coverage.

**Review by Victoradukwu — APPROVED** [review]
LGTM

**Review by espoirMur — APPROVED** [review]

**Review by CryceTruly — APPROVED** [review]

**Review by bl4ck4ndbr0wn — APPROVED** [review]
LGTM

**Pull request merged**